cdap-master/src/main/java/co/cask/cdap/data/tools/DataMigration.java

```diff
@@ -77,53 +83,68 @@ private String getDescription() {
     }
   }
 
-  public void doMain(String[] args) throws Exception {
+  public boolean doMain(String[] args) throws Exception {
     System.out.println(String.format("%s - version %s.", getClass().getSimpleName(), ProjectInfo.getVersion()));
     System.out.println();
 
     if (args.length < 1) {
       printHelp();
-      return;
+      return false;
     }
 
     Action action = parseAction(args[0]);
     if (action == null) {
       System.out.println(String.format("Unsupported action : %s", args[0]));
       printHelp(true);
-      return;
+      return false;
     }
 
-    try {
-      CConfiguration cConf = CConfiguration.create();
-      Configuration hConf = HBaseConfiguration.create();
-
-      Injector injector = Guice.createInjector(
-        new ConfigModule(cConf, hConf),
-        new LocationRuntimeModule().getDistributedModules(),
-        new AbstractModule() {
-          @Override
-          protected void configure() {
-            bind(HBaseTableUtil.class).toProvider(HBaseTableUtilFactory.class);
-            bind(QueueAdmin.class).to(HBaseQueueAdmin.class).in(Singleton.class);
-            install(new FactoryModuleBuilder()
-                      .implement(DatasetDefinitionRegistry.class, DefaultDatasetDefinitionRegistry.class)
-                      .build(DatasetDefinitionRegistryFactory.class));
-          }
-        });
-
-      switch (action) {
-        case METRICS:
-          migrateMetricsData(injector);
-          break;
-        case HELP:
-          printHelp();
-          break;
+    if (action.equals(Action.METRICS)) {
+      if (args.length > 2) {
+        System.out.println("invalid number of arguments");
+        printHelp(true);
+        return false;
+      } else if (args.length == 2) {
+        if (args[1].equals(KEEP_OLD_METRICS_DATA)) {
+          keepOldMetricsData = true;
+        } else {
+          System.out.println("invalid argument, expected argument " + KEEP_OLD_METRICS_DATA);
+          printHelp(true);
+          return false;
+        }
       }
-    } catch (Exception e) {
-      System.out.println(String.format("Failed to perform action '%s'. Reason: '%s'.", action, e.getMessage()));
-      e.printStackTrace(System.out);
-      throw e;
     }
+
+    if (isTest) {
```

Comment:
with doing as in the above comment - you can extract method that will take in args and return MigrationAction instance. Then you'll test that method - no need to hack with isTest field, etc.